cloud/sts.go

```diff
@@ -77,7 +79,44 @@ func (s *stsManagerImpl) AssumeRole(ctx context.Context, taskID string, opts Ass
 	if t == nil {
 		return AssumeRoleCredentials{}, errors.New("task not found")
 	}
-	externalID := createExternalID(t)
+	p, err := model.GetProjectRefForTask(ctx, taskID)
+	if err != nil {
+		return AssumeRoleCredentials{}, errors.Wrapf(err, "getting project ref for task")
+	}
+	if p == nil {
+		return AssumeRoleCredentials{}, errors.New("project ref not found")
+	}
+	externalID := createExternalID(t, p)
+	creds, err := s.assumeRole(ctx, externalID, opts)
+	if err != nil {
+		externalID = createExternalID(t, nil)
+		var fallbackErr error
+		creds, fallbackErr = s.assumeRole(ctx, externalID, opts)
+		if fallbackErr != nil {
+			return AssumeRoleCredentials{}, errors.Wrapf(err, "assuming role, fallback error: '%v'", fallbackErr)
+		}
+		// Only log if the fallback succeeded.
+		grip.Debug(message.Fields{
```

Comment:
can we remove this or do we plan on removing it after some criteria in the future?